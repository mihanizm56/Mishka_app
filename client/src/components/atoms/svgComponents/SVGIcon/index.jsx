//
import React from "react";
import "./SVGIcon.css";
import MediaQuery from "react-responsive";
import { SVGComponent } from "..";
import { getSmallImage, getMiddleImage, getBigImage } from "../../../../utils";
import {
	BIG_MEDIA_SIZE,
	MIDDLE_MEDIA_SIZE_FROM,
	MIDDLE_MEDIA_SIZE_TO,
	SMALL_MEDIA_SIZE_FROM,
	SMALL_MEDIA_SIZE_TO,
} from "../../../../constants";

type SVGIconProps = {
	icon: string,
};

export const SVGIcon = ({ icon }: SVGIconProps) => {
	const smallIcon = getBigImage(`icon-${icon}-big`);
	const middleIcon = getMiddleImage(`icon-${icon}-middle`);
	const bigIcon = getSmallImage(`icon-${icon}-small`);
	return (
		<>
			<MediaQuery minWidth={BIG_MEDIA_SIZE}>
				<SVGComponent path={smallIcon} />
			</MediaQuery>
			<MediaQuery minWidth={MIDDLE_MEDIA_SIZE_FROM} maxWidth={MIDDLE_MEDIA_SIZE_TO}>
				<SVGComponent path={middleIcon} />
			</MediaQuery>
			<MediaQuery minWidth={SMALL_MEDIA_SIZE_FROM} maxWidth={SMALL_MEDIA_SIZE_TO}>
				<SVGComponent path={bigIcon} />
			</MediaQuery>
		</>
	);
};
