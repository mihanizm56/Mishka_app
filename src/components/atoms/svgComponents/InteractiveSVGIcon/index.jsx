import React from "react";
import { SVGIcon } from "../";
import { getClass } from "../../../../utils";
import "./InteractiveSVGIcon.css";

export const InteractiveSVGIcon = ({ icon, handleClick, noOpacityChange }) => {
	return (
		<div className={getClass({ initialClass: "interactive-icon-wrapper", noOpacityChange: noOpacityChange })}>
			<SVGIcon icon={icon} />
			<div className="click-wrapper" onClick={handleClick} />
		</div>
	);
};

InteractiveSVGIcon.defaultProps = {
	icon: "",
	handleClick: () => {},
};
