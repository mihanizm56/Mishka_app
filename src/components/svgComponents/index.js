export * from "./InteractiveSVGIcon";
export * from "./SVGComponent";
export * from "./SVGForPage";
