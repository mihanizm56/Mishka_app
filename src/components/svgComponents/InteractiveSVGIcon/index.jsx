import React from "react";
import { SVGForPage } from "../";
import { getClass } from "../../../helpers";
import "./InteractiveSVGIcon.css";

export const InteractiveSVGIcon = ({ icon, handleClick, noOpacityChange }) => {
	return (
		<div className={getClass({ initialClass: "interactive-icon-wrapper", noOpacityChange: noOpacityChange })}>
			<SVGForPage icon={icon} />
			<div className="click-wrapper" onClick={handleClick} />
		</div>
	);
};

InteractiveSVGIcon.defaultProps = {
	icon: "",
	handleClick: () => {},
};
