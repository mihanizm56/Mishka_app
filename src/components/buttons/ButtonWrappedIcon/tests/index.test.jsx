import React from 'react';
import { shallow } from 'enzyme';
import { ButtonWrappedIcon } from '../';
import { InteractiveSVGIcon } from '../../../svgComponents'

describe('ButtonWrappedIcon test', () => {
    const mockClick = jest.fn();
    const testString = 'test string';
    const customWidth = '100px';
    const customHeight = '100px';
    const defaultString =  "no icon"

    beforeEach(() => {
        jest.clearAllMocks();
    });

    describe('test render', () => {
        describe('render with all props', () => {
            const component = shallow(<ButtonWrappedIcon  width={customWidth} height={customHeight} icon={testString} handleClick={mockClick} noOpacityChange />);
            
            it('render wrapper', () => {
                expect(component.find('.button-wrapped-icon').length).toEqual(1);
            });
            it('render component', () => {
                expect(component.find(InteractiveSVGIcon).length).toEqual(1);
            });
        })
        describe('render without any props', () => {
            const component = shallow(<ButtonWrappedIcon />);

            it('render wrapper', () => {
                expect(component.find('.button-wrapped-icon').length).toEqual(1);
            });
            it('render component', () => {
                expect(component.find(InteractiveSVGIcon).length).toEqual(1);
            });
        })
    })

    describe('test props', () => {
        describe('with all props on component', () => {
            const component = shallow(<ButtonWrappedIcon  width={customWidth} height={customHeight} icon={testString} handleClick={mockClick} noOpacityChange />);

            it('check style prop on wrapper', () => {
                expect(component.find('.button-wrapped-icon').props().style).toEqual({width: customWidth,height: customHeight});
            });
            it('check onClick prop on wrapper', () => {
                expect(component.find('.button-wrapped-icon').props().onClick).toEqual(mockClick);
            });
            it('check handleClick prop on InteractiveSVGIcon', () => {
                expect(component.find(InteractiveSVGIcon).props().handleClick).toEqual(mockClick);
            });
            it('check icon prop', () => {
                expect(component.find(InteractiveSVGIcon).props().icon).toEqual(testString);
            });
            it('check noOpacityChange prop', () => {
                expect(component.find(InteractiveSVGIcon).props().noOpacityChange).toBeTruthy();
            });
        })
        describe('without any props', () => {
            const component = shallow(<ButtonWrappedIcon />);

            it('snapshot test', () => {
                expect(component).toMatchSnapshot();
            });
        })
    })

    describe('action of component', () => {
        const component = shallow(<ButtonWrappedIcon handleClick={mockClick} />);

        it('check callback on click', () => {
            component.find('.button-wrapped-icon').simulate('click');
            expect(mockClick).toHaveBeenCalled();
        });
    })
})
