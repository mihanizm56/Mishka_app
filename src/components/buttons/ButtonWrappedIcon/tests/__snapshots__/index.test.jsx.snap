// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWrappedIcon test test props without any props snapshot test 1`] = `
<div
  className="button-wrapped-icon"
  onClick={[Function]}
  style={null}
>
  <InteractiveSVGIcon
    handleClick={[Function]}
    icon="no icon"
    noOpacityChange={false}
  />
</div>
`;
